app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/UrlProcessorTypeMatcher.java

```diff
@@ -0,0 +1,36 @@
+package com.github.kulebin.myfoursquareapp.dataSource;
```

Comment:
Too complicated. Just use processor type as a param for loadData method
UriMatcher is good approach when you use ContentProvider and Content Resolver, when you use some notifiers and so on.
In our case - we khow which processor to use when we call loadData. So we do extra steps to get processorType.
And we also need to modify this class if we add new api calls, and even if just url changes!!!

We can add a result class to ProcessorType, and try to match it from IOnResultCallback<Result> type using reflection. But for now it's extra efforts.